src/server.py

```diff
@@ -20,37 +24,38 @@ def server():
         conn, addr = server.accept()
         buffer_length = 32
         message_complete = False
-        full_mes = b''
+        request = b''
         while not message_complete:
-            partial_mes = conn.recv(buffer_length)
-            full_mes += partial_mes
-            if len(partial_mes) < buffer_length:
+            buffer_req = conn.recv(buffer_length)
+            request += buffer_req
+            if len(buffer_req) < buffer_length:
                 message_complete = True
-                full_mes_decoded_to_unicode = full_mes.decode('utf8')
-        print(u'request:\r\n', full_mes_decoded_to_unicode)
-        response_mes, uri = response_decision(full_mes)
-        conn.sendall(response_mes)
+        response = response_decision(request)
+        conn.sendall(response)
         conn.close()
         server.close()
 
 
-def response_ok():
+def response_ok(body, mime=None):
     """Generate response_ok."""
+    try:
+        body = body.encode('utf-8')
+    except (UnicodeDecodeError, AttributeError):
```

Comment:
Interesting approach.  This will throw a UnicodeDecodeError when you have bytes already in Python2 and the AttributeError if you have bytes already in Python3.